azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -153,189 +263,328 @@ public long getLastCleanerThreadCheckTime() {
+    for (Pair<ExecutionReference, ExecutableFlow> ref : queuedFlows) {
+      queuedFlowMap.put(ref.getSecond().getExecutionId(), ref);
+    }
+  }
+
+  /**
+   * Gets a list of all the active (running, non-dispatched) executions for a
+   * given project and flow {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorManagerAdapter#getRunningFlows(int,
+   *      java.lang.String)
+   */
   @Override
   public List<Integer> getRunningFlows(int projectId, String flowId) {
     ArrayList<Integer> executionIds = new ArrayList<Integer>();
-    for (Pair<ExecutionReference, ExecutableFlow> ref : runningFlows.values()) {
+    getRunningFlowsHelper(projectId, flowId, executionIds,
+      queuedFlowMap.values());
+    getRunningFlowsHelper(projectId, flowId, executionIds,
+      runningFlows.values());
+    return executionIds;
+  }
+
+  /* Helper method for getRunningFlows */
+  private void getRunningFlowsHelper(int projectId, String flowId,
+    ArrayList<Integer> executionIds,
+    Collection<Pair<ExecutionReference, ExecutableFlow>> collection) {
+    for (Pair<ExecutionReference, ExecutableFlow> ref : collection) {
       if (ref.getSecond().getFlowId().equals(flowId)
-          && ref.getSecond().getProjectId() == projectId) {
+        && ref.getSecond().getProjectId() == projectId) {
         executionIds.add(ref.getFirst().getExecId());
       }
     }
-    return executionIds;
   }
 
+  /**
+   *
+   * {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorManagerAdapter#getActiveFlowsWithExecutor()
+   */
+  @Override
+  public List<Pair<ExecutableFlow, Executor>> getActiveFlowsWithExecutor()
+    throws IOException {
+    List<Pair<ExecutableFlow, Executor>> flows =
+      new ArrayList<Pair<ExecutableFlow, Executor>>();
+    getActiveFlowsWithExecutorHelper(flows, queuedFlowMap.values());
+    getActiveFlowsWithExecutorHelper(flows, runningFlows.values());
+    return flows;
+  }
+
+  /* Helper method for getActiveFlowsWithExecutor */
+  private void getActiveFlowsWithExecutorHelper(
+    List<Pair<ExecutableFlow, Executor>> flows,
+    Collection<Pair<ExecutionReference, ExecutableFlow>> collection) {
+    for (Pair<ExecutionReference, ExecutableFlow> ref : collection) {
+      flows.add(new Pair<ExecutableFlow, Executor>(ref.getSecond(), ref
+        .getFirst().getExecutor()));
+    }
+  }
+
+  /**
+   * Checks whether the given flow has an active (running, non-dispatched)
+   * executions {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorManagerAdapter#isFlowRunning(int,
+   *      java.lang.String)
+   */
   @Override
   public boolean isFlowRunning(int projectId, String flowId) {
-    for (Pair<ExecutionReference, ExecutableFlow> ref : runningFlows.values()) {
+    boolean isRunning = false;
+    isRunning =
+      isRunning
+        || isFlowRunningHelper(projectId, flowId, queuedFlowMap.values());
+    isRunning =
+      isRunning
+        || isFlowRunningHelper(projectId, flowId, runningFlows.values());
+    return false;
```

Comment:
nope is returns true when isFlowRunningHelper returns true i.e. when project-flow ids are present in either queuedFlowMap or runningFlows.